### ISSUE TITLE
When support VP9 ?

### ISSUE THREAD

**Issue by zhangalex** [Kurento/bugtracker#239]
It is likely that KMS doesn't support VP9 for now. Do you have any plan on VP9 ?

**Comment by LeoColomb** [comment]
Or even the new AV1 (the successor of VP9). 👍 
These supports need at least a Gstreamer update (which seems to be [on the roadmap](http://www.kurento.org/blog/kurento-67-moving-forward) 🙌).

**Comment by micaelgallego** [comment]
One of our main priorities is update GStreamer to the latest version (1.14)
released some weeks ago. Also we use other libraries that we plan to update
also.

When we update all those libraries, we will start testing new codecs like
VP9 and AV1.

**Comment by LeoColomb** [comment]
@zhangalex Any reason you are closing the issue? 🙂

**Comment by micaelgallego** [comment]
The question has been answered ;)

But we can maintain it open to track progress... at least until better issue is created for that.

**Comment by zhangalex** [comment]
@micaelgallego  Yes, it has been answered. haha.

**Comment by micaelgallego** [comment]
I can't tell you when VP9 support will be available. It depends on how much time we spend updating GStreamer, testing VP9 support on that, testing compatibility with Chrome, etc. 

Just for curiosity, why do you want VP9 support? Only for reducing bandwidth usage?

**Comment by zhangalex** [comment]
@micaelgallego   Yes, because our project is running in a private network (it cannot access internet), and the bandwidth in some area is not stable, so we need reduce bandwidth.

**Comment by zhangalex** [comment]
Is there any progress on this?

**Comment by Ezanor** [comment]
@micaelgallego 

Any news on that ?

**Comment by broderix** [comment]
up

**Comment by micaelgallego** [comment]
We are in the process to update Kurento Media Server to GStreamer 1.14. We will take a look to VP9 support after that

**Comment by j1elo** [comment]
Closing because this is not a bug. Also, the update to GStreamer 1.14 got stalled for some time although 2020 seems like it will see the light.

In any case, top priority to Kurento is keep compatibility with Chrome and Firefox, that is already enough work for our small team, and it doesn't make much sense working to support AV1 if Chrome doesn't support it either...

VP9 will be considered as a "nice to have" feature when porting to GStreamer 1.14 (and possibly 1.16) is finally done.

**Comment by alfredonodo** [comment]
Hi,
I wonder to know if now kurento supports VP9 video codec.
Thank you

**Comment by kfbixc** [comment]
any progress?

**Comment by robregonm** [comment]
I think that it's a better idea to focus on **AV1**, since it is supported by Safari, Chrome and Firefox, and compression rates are much better than VP9.
Notice that VP9 is not currently supported by Safari

**Comment by alfredonodo** [comment]
@robregonm safari supports VP9 codec on WebRTC since iOS14 https://webkit.org/blog/10929/release-notes-for-safari-technology-preview-110/.
The main problem with AV1 and with VP9 too, is the missing hardware encoding and decoding capabilities (VP9 it is a lot ahead on this https://en.wikipedia.org/wiki/VP9#Hardware_device_support).

**Comment by pachainti** [comment]
@j1elo any news?

**Comment by j1elo** [comment]
Kurento will not support VP9 or AV1 in the foreseeable future; we're working to add this directly to OpenVidu; if there is enough commercial interest, we'd possibly fund progress on upgrading Kurento's GStreamer fork, which in turn unblocks the addition of newer codecs.

**Comment by themightyoarfish** [comment]
I'm assuming this won't happen? Bummer.
What specifically makes this difficult? ffmpeg and gstreamer both support vp9 encoding and payloading for RTP for instance.
